test/functional/visual/ctrl_o_spec.lua

```diff
@@ -0,0 +1,23 @@
+local helpers = require('test.functional.helpers')(after_each)
+local assert_alive = helpers.assert_alive
+local clear = helpers.clear
+local eq = helpers.eq
+local expect = helpers.expect
+local feed = helpers.feed
+local meths = helpers.meths
+
+describe('select-mode Ctrl-O', function()
+  before_each(clear)
+
+  it("doesn't cancel Ctrl-O mode when processing event", function()
+    feed('iHello World<esc>gh<c-o>')
+    eq({mode='vs', blocking=false}, meths.get_mode()) -- fast event
+    assert_alive()  -- causes K_EVENT key
```

Comment:
I would spell out the request directly in this case, as the purpose is not to `assert_alive` and also the implementation of the latter could change.

Reply:
Err... What request?

Reply:
yes, that works :)